File: 979-di-string-match/di-string-match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        low = 0
        high = len(s)
        output = [None] * len(s)

        for i in range(len(s)):
            char = s[i]
            if (char == "I"):
                output[i] = low
                low += 1
            else:
                output[i] = high
                high -= 1

        output.append(low)
        return output

File: 979-di-string-match/di-string-match.js
var diStringMatch = function(s) {
    let i = 0
    let j = s.length
    let output = new Array()

    for (let char of s) {
        if (char == "I") {
            output.push(i)
            i += 1
        }
        else {
            output.push(j)
            j -= 1
        }
    }

    output.push(i)
    return output
};